From 5a28cb783a33391476839a5b904f0eab7fb9b13d Mon Sep 17 00:00:00 2001
From: Max Ihlenfeldt <max@igalia.com>
Date: Fri, 22 Mar 2024 10:43:47 +0000
Subject: [PATCH] Fix "constexpr variable must be initialized by a constant
 expression"

clang 17 still has some problems with constexpr strings. We need to
change the problematic expressions to be const instead.

Upstream-Status: Inappropriate [specific to older versions of clang]
Signed-off-by: Max Ihlenfeldt <max@igalia.com>
---
 .../autofill/core/browser/data_model/autofill_i18n_api.h      | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/components/autofill/core/browser/data_model/autofill_i18n_api.h b/components/autofill/core/browser/data_model/autofill_i18n_api.h
index fa1a06bdad..8e1fe01bfc 100644
--- a/components/autofill/core/browser/data_model/autofill_i18n_api.h
+++ b/components/autofill/core/browser/data_model/autofill_i18n_api.h
@@ -16,8 +16,8 @@ namespace autofill::i18n_model_definition {
 // Country code that represents autofill's legacy address hierarchy model as
 // stored `kAutofillModelRules`. As a workaround for GCC we declare the
 // std::string constexpr first.
-constexpr inline std::string kLegacyHierarchyCountryCodeString{"XX"};
-constexpr AddressCountryCode kLegacyHierarchyCountryCode =
+const inline std::string kLegacyHierarchyCountryCodeString{"XX"};
+const AddressCountryCode kLegacyHierarchyCountryCode =
     AddressCountryCode(kLegacyHierarchyCountryCodeString);

 // Creates an instance of the address hierarchy model corresponding to the
